### DISCUSSION TITLE
An open source CollectionView control that works on Windows ?

### DISCUSSION THREAD

**Discussion by BeepBeepBopBop** [dotnet/maui#25110] [General]
Hello, I have been using MAUI for over a year now and though I like the technology I never was able to work on Windows with the CollectionView control. There are so many issues that I don't think it's even worth mentioning them at this point, let's just stay if you have tried to create a professional looking app you know it's not possible without a bunch of workarounds.

So far I have been using workarounds but it's a pain to maintain and I have decided to tackle the issue and create my own implementation of CollectionView based on StackLayout + GestureRecognizers + StateManager basically. 

It's a lot of work though, and I am not trying to re-invent the wheel so I would like to hear from you, have you found some open-source implementation of a CollectionView controls that works correctly on Windows ? I know about this one but it's for mobile only 
 https://github.com/roubachof/Sharpnado.CollectionView/tree/main/Maui

Thanks for your input

**Comment by MartyIX** [comment]
Overall, I wonder if it is not worth it to rather try to fix it in MAUI itself. With Appium tests, one can be quite sure that it will keep on working in newer MAUI releases (as opposed to workarounds that migth stop working or custom implementation that is costly to develop).

I don't use CollectionView really but some of my PRs were merged so if something is not clear in the contribution process (or actual implementation process), I think I can help with that.

Edit: For example, there is https://github.com/dotnet/maui/pull/24830 and a great contribution would be to test it on Windows.

**Comment by BeepBeepBopBop** [comment]
Thanks for your input, I do agree with you. I am still learning MAUI and XAML and don't have confidence nor much time to help with the code at this time but I will definitely consider it in a near future. 

In the end I decided to implement my own CollectionView control with the intention of implementing a basic functional one in the least possible hacky fashion, taking benefits of the built-in features I have found to be stable. If anybody comes here looking for some help you can see it [here](https://github.com/BeepBeepBopBop/CustomCollectionView)
